File: examples/simple/ThriftConsumer.rb
#!/usr/bin/env ruby
$LOAD_PATH.push '../../lib/rb/'
$LOAD_PATH.push '../gen-rb/'
require 'consumer'
require 'example_types'

c = Jp::ThriftConsumer.new 'thrift', ExampleData do |message|
	print "I consumed:\n%s\n" % message.inspect
	true # must return something that evalutes to true in a boolean context, or purge() isn't called.
end
c.run

File: examples/simple/ThriftConsumer.php
#!/usr/bin/php
<?php

set_time_limit(0);

$jpRoot = dirname(__FILE__) . '/../../';
set_include_path(get_include_path() . PATH_SEPARATOR . $jpRoot . 'lib/php' . PATH_SEPARATOR . $jpRoot . 'gen-php/jp');
$GLOBALS['THRIFT_ROOT'] = 'Thrift';
require_once('jp/consumer/Thrift.php');
require_once('jp/worker/Interface.php');
require_once(dirname(__FILE__) . '/../gen-php/example/example_types.php');

class ThriftQueueWorker implements jp_worker_Interface {
	public function processItem($message) {
		echo 'I consumed: ' . print_r($message, true);
		return true;
	}
}

$processor = new jp_consumer_Thrift('thrift', array('poll_interval' => 200), new ThriftQueueWorker(), 'ExampleData');
$processor->run();
